ts-sdk/whirlpool/src/createPool.ts

```diff
@@ -242,12 +243,19 @@ export async function createConcentratedLiquidityPoolInstructions(
         startTickIndex: tickArrayIndexes[i],
       }),
     );
-    stateSpace += getTickArraySize();
+    stateSpaces.push(getTickArraySize());
   }
 
-  const nonRefundableRent = await rpc
-    .getMinimumBalanceForRentExemption(BigInt(stateSpace))
-    .send();
+  const nonRefundableRents: Lamports[] = await Promise.all(
+    stateSpaces.map(async (space) => {
+      const rentExemption = await calculateMinimumBalance(rpc, space);
```

Comment:
This is still doing the same thing where it fetches the rent 5 (or 6 times).

The way I would do it. Change the function to:

```
export function calculateMinimumBalance(
  rent: Rent,
  dataSize: number,
): Lamports {
```

Fetcht he rent sysvar at the beginning of this function and then instead of keeping track of `stateSpace: Array<bigint>` you can just straight up keep track of `estInitializationCost: Lamports`.

And then instead of doing `stateSpace.push(...)` you can just do estInitializationCost += calculateMinimumBalance(rent, ...)